theme/theme.go

```diff
@@ -275,6 +285,9 @@ func (t *builtinTheme) initFonts() {
 }
 
 func (t *builtinTheme) Color(n fyne.ThemeColorName, v fyne.ThemeVariant) color.Color {
+	if t.variant != variantNameUserPreference {
+		v = t.variant
+	}
```

Comment:
This does not seem to be related to the text refactor, and it would change the current behavior of Light and Dark themes (extending using Light or Dark themes is equal to extending DefaultTheme, after this change this will not true anymore). `currentVariant ()` function would not have much sense after this change. I understand this change is the right way to go, but maybe this should be in another PR? 
Also, I wonder if the theme API could decide if the a theme is desired to handle the variants or the variant handles the themes, because now it is a bit confusing. In application examples, the theme handle the variants, but in the core with the use of the Light and Dark themes works vice versa.

Reply:
Without this change the theme lookups do not work as expected.
Ultimately I got the implementation wrong the first time round.
I don't see what has been broken, what this does is ensure that `LightTheme()` and `DarkTheme()` ignore the current user preference.

A theme can feel free to ignore the variant, if that is desired. I don't understand what you mean about core themes having this backwards. LightTheme and DarkTheme are (historical) conveniences for apps that want to turn off the automatic switching. We could deprecate them and force developers to implement custom themes, but it seems overly harsh.

Reply:
> I don't see what has been broken, what this does is ensure that LightTheme() and DarkTheme() ignore the current user preference.

Yes, that is right, it is just that before if you created a custom theme based on the DarkTheme or LightTheme it didn't matter, because both of them acted like the DefaultTheme.

> A theme can feel free to ignore the variant, if that is desired. I don't understand what you mean about core themes having this backwards. LightTheme and DarkTheme are (historical) conveniences for apps that want to turn off the automatic switching. We could deprecate them and force developers to implement custom themes, but it seems overly harsh.

Completely agreed. But what I mean is that usually you use a new theme for each variant (just like the DarkTheme and LightTheme), but the current Theme API encourages you to define one single theme that handles the variants. That leads to have functions like `theme.Dark (). Color (name, theme.VariantLight)` that evidently has no much sense. So, IMHO, we should rather change the theme interface Color method to accept only the color name without using the variant parameter (and encourage to always define a theme per variant) or remove DarkTheme and LightTheme and use the DefaultTheme and force the variant selection ( in that way we are always encouraging to define a single theme that handles all the variants). The first option should be the right direction as that is how usually others toolkits handle this usecase.

Reply:
> But what I mean is that usually you use a new theme for each variant

This is not the case. In 2.0 a theme defines how the app will look and the variant represents the user's preference (often derived from the OS settings). Apps should certainly not be encouraged to decide if they want to look light or dark, them should support both (with one theme).

If there is confusion I can deprecate `LightTheme()` and `DarkTheme()` as really they should not be needed by modern theme creation.

Reply:
As you said before, "LightTheme and DarkTheme are (historical) conveniences for apps that want to turn off the automatic switching.", so deprecating them without offer a wait to force a specific variant seems not to be good (final users may want to use DarkTheme only in that application instead of activating it in the whole system). So if they are deprecated we should consider adding a method to override the default variant, just like Android, iOS, Flutter and other toolkits allow the developers to do:
https://developer.android.com/guide/topics/ui/look-and-feel/darktheme#change-themes
https://developer.apple.com/documentation/uikit/appearance_customization/supporting_dark_mode_in_your_interface/choosing_a_specific_interface_style_for_your_ios_app

But obviously this should be discussed in detail in other PR related to the theme.

Reply:
I am confused here sorry - you seem to be suggesting end user configuration, but they don't have access to the APIs. 
The theme API allows app developers to ignore the user preference of variant by adding their own theme - I don't want to add this to the app API as it is not something we should encourage.

Reply:
> I am confused here sorry - you seem to be suggesting end user configuration, but they don't have access to the APIs.

End user configuration is the goal but to accomplish it developers would need to access to the APIs. I am aware that this is entirely possible at current state, however if I define a theme that handles Light and Dark variants and I want to let end users choose if they want to follow their system configuration or force to a specific variant, then I have to define two different themes to handle it (or add some logic to my current theme to handle the special case), instead of simply override that variant behavior as Android, iOS and others allow.

Reply:
We must be talking at crossed purposes - this is making no sense to me.
Current state:
* Default is system configuration (light or dark)
* User can choose (through fyne_settings or FYNE_THEME) to override this and specify light or dark (or fall back to system default)
* All themes are told which the current variant is in the `Color` call - and they can choose to ignore it.

An app that calls `Settings.SetTheme(DarkTheme())` is overriding all of these system and user settings, because the app wants to force a specific look.

On which part are we not managing to agree on?